src/gluonts/dataset/split.py

```diff
@@ -73,130 +73,122 @@
 from gluonts.dataset.field_names import FieldName
 
 
-class TimeSeriesSlice(pydantic.BaseModel):
+def to_positive_slice(slc: slice, length: int) -> slice:
     """
-    Like DataEntry, but all time-related fields are of type pd.Series and is
-    indexable, e.g `ts_slice['2018':]`.
+    Returns an equivalent slice with positive bounds, given the
+    length of the sequence it will apply to.
     """
+    start, stop = slc.start, slc.stop
+    if start is not None and start < 0:
+        start += length
+        assert start >= 0
+    if stop is not None and stop < 0:
+        stop += length
+        assert stop >= 0
+    return slice(start, stop, slc.step)
 
-    class Config:
-        arbitrary_types_allowed = True
 
-    target: pd.Series
-    item_id: Optional[str] = None
+def to_integer_slice(slc: slice, start: pd.Period) -> slice:
+    """
+    Returns an equivalent slice with integer bounds, given the
+    start timestamp of the sequence it will apply to.
+    """
+    start_is_int = isinstance(slc.start, (int, type(None)))
+    stop_is_int = isinstance(slc.stop, (int, type(None)))
 
-    feat_static_cat: List[int] = []
-    feat_static_real: List[float] = []
+    if start_is_int and stop_is_int:
+        return slc
 
-    feat_dynamic_cat: List[pd.Series] = []
-    feat_dynamic_real: List[pd.Series] = []
+    if isinstance(slc.start, pd.Period):
+        start_offset = (slc.start - start).n
+        assert start_offset >= 0
+    elif start_is_int:
+        start_offset = slc.start
+    else:
+        raise ValueError()
 
-    @classmethod
-    def from_data_entry(
-        cls, item: DataEntry, freq: Optional[str] = None
-    ) -> "TimeSeriesSlice":
-        if freq is None:
-            freq = item["start"].freq
+    if isinstance(slc.stop, pd.Period):
+        stop_offset = (slc.stop - start).n + 1
+        assert stop_offset >= 0
+    elif stop_is_int:
+        stop_offset = slc.stop
+    else:
+        raise ValueError()
 
-        index = pd.period_range(
-            start=item["start"], freq=freq, periods=len(item["target"])
-        )
+    return slice(start_offset, stop_offset)
 
-        feat_dynamic_cat = [
-            pd.Series(cat, index=index)
-            for cat in list(item.get("feat_dynamic_cat", []))
-        ]
 
-        feat_dynamic_real = [
-            pd.Series(real, index=index)
-            for real in list(item.get("feat_dynamic_real", []))
-        ]
+def slice_data_entry(
+    entry: DataEntry, slc: slice, prediction_length: int = 0
+) -> DataEntry:
+    slc = to_positive_slice(
+        to_integer_slice(slc, entry[FieldName.START]),
+        len(entry[FieldName.TARGET]),
+    )
 
-        feat_static_cat = list(item.get("feat_static_cat", []))
+    if slc.stop is not None:
+        slc_extended = slice(slc.start, slc.stop + prediction_length, slc.step)
+    else:
+        slc_extended = slc
 
-        feat_static_real = list(item.get("feat_static_real", []))
+    sliced_entry = entry.copy()
```

Comment:
I like this notion better

```suggestion
    sliced_entry = dict(entry)
```